xarray/backends/rasterio_.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+
+try:
+    import rasterio
+except ImportError:
+    rasterio = False
+
+from .. import Variable, DataArray
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin
+from ..core import indexing
+from ..core.pycompat import OrderedDict
+
+from .common import AbstractDataStore
+
+__rio_varname__ = 'raster'
+
+
+class RasterioArrayWrapper(NDArrayMixin):
+    def __init__(self, ds):
+        self._ds = ds
+        self.array = ds.read()
+
+    @property
+    def dtype(self):
+        return np.dtype(self._ds.dtypes[0])
+
+    def __getitem__(self, key):
+        if key == () and self.ndim == 0:
+            return self.array.get_value()
+        return self.array[key]
+
+
+class RasterioDataStore(AbstractDataStore):
+    """Store for accessing datasets via Rasterio
+    """
+    def __init__(self, filename, mode='r'):
+
+        with rasterio.Env():
+            self.ds = rasterio.open(filename, mode=mode, )
+
+            # Get coords
+            nx, ny = self.ds.width, self.ds.height
+            x0, y0 = self.ds.bounds.left, self.ds.bounds.top
+            dx, dy = self.ds.res[0], -self.ds.res[1]
+
+        self.coords = {'y': np.arange(start=y0, stop=(y0 + ny * dy), step=dy),
+                       'x': np.arange(start=x0, stop=(x0 + nx * dx), step=dx)}
+
+        # Get dims
+        if self.ds.count >= 1:
+            self.dims = ('band', 'y', 'x')
+            self.coords['band'] = self.ds.indexes
+        else:
+            raise ValueError('unknown dims')
+
+        self._attrs = OrderedDict()
+        for attr_name in ['crs', 'transform', 'proj']:
+            try:
+                self._attrs[attr_name] = getattr(self.ds, attr_name)
+            except AttributeError:
+                pass
+
+        self.coords = _try_to_get_latlon_coords(self.coords, self._attrs)
```

Comment:
Thanks. This now pulls in lat/long if the projection was successful, otherwise it just returns the basic x/y indices. Should we also be returning the original projection coords? (currently they are lost).